### ISSUE TITLE
Animation results in only the last frame being shown statically

### ISSUE THREAD

**Issue by gsaurabhr** [jwkvam/celluloid#15]
```
def generate_frame(a, s, d, j, g, t_win, f):
    plt.clf()
    plt.plot(range(a), [i*i for i in range(a)])
    return f

plt.ioff()
fig = plt.figure(figsize=(8, 6))
camera = Camera(fig)
for i in range(10):
    generate_frame(i*10, 0, 0, 0, 0, 0, f=fig)
    camera.snap()
animation = camera.animate()
HTML(animation.to_html5_video())
```

This does not work. The resultant animation shows only the final frame.

The extra parameters in the generate_frame function are leftovers from my real function, but should not matter.

**Comment by Tinggaard** [comment]
I've had the same problem myself, fixed it by not using `plt.subplots()`, but simply writing to the "root", by calling `plt.function()` as opposed to `fig.function()`.
```shell
Package         Version
--------------- -------
celluloid       0.2.0  
matplotlib      3.2.1  
numpy           1.18.2 
```

**Comment by flavio-martinelli** [comment]
Sorry @Tinggaard I don't understand your answer, OP never used `plt.subplots( )`. What do you mean by writing to the 'root' ? Can you provide a simple code that solves OP's issue?

I am having the same issue myself

**Comment by Tinggaard** [comment]
@flavio-martinelli I did not solve the exact problem OP had, I had one simmilar and solved it by calling `plt.plot()` every time in my loop, as I were not calling it beforehand. 

I were trying to update an image using blit, if I recall correctly, I were trying to achieve it, by calling `plt.subplots()`, and then updating the returned figure.
I ended up running it like this:
```python3
import numpy as np
import matplotlib.pyplot as plt
from celluloid import Camera

# setup
class Foo:
    def __init__(self, maze: np.ndarray):
        self.mz = maze

        self.cam = Camera(plt.figure())
        implot = plt.imshow(self.mz, interpolation='nearest', aspect='equal', vmin=0, vmax=255, cmap="RdBu")
        implot.set_cmap('hot')
        plt.axis('off')
        self.cam.snap()


    # calling function after change to self.mz
    def plot_changes(self):
        plt.imshow(self.mz)
        self.cam.snap()

    # function to wrap it all up into a video
    def save(self, destination):
        anim = self.cam.animate(blit=True, interval=30)
        anim.save(destination)
```
I cannot remember exactly what I were doing earlier, but I think it was something along the lines of changing `Camera()`, to take `plt.figure()`, instead of `plt.subplots()`. That is also what OP is doing, so I have no idea where the error originates.

But this is the code I am using now, which works for me.

**Comment by flavio-martinelli** [comment]
Thanks for your detailed reply!

I actually fixed the last frame problem by removing `plt.clf()` and adding `plt.draw()` just before every call to `camera.snap()`. Not really sure what is happening there, but on a Jupyter notebook seems the way to go.
